bin/init_helm.sh

```diff
@@ -45,12 +45,12 @@ export ISTIO_BIN=${ISTIO_BIN:-${GOPATH}/bin}
 export GOARCH=${GOARCH:-'amd64'}
 
 # Determine the OS. Matches logic in the Makefile.
-LOCAL_OS="`uname`"
+LOCAL_OS=${OSTYPE}
```

Comment:
where does ${OSTYPE} come from?  I don't see this in the repo via grep...  I do see LOCAL_OS in the Makefile.

Could you expand on what problem you are solving with this change?

Reply:
As per @rshriram note, deploying helm on mac was not working correctly. With this change MAC os detection works and correct binary gets retrieved from the helm site.

Reply:
I suggest you run on your MAC `echo $OSTYPE`

Reply:
On Mac:
```
STDAKE-M-26T1:~ sdake$ echo $OSTYPE
darwin17
```

On Linux:
```
sdake@falkor-08:~/go/src/istio.io/istio$ echo $OSTYPE
linux-gnu
```

Wasn't aware this was a standard environment variable.